core/unit_test/TestNonTrivialScalarTypes.hpp

```diff
@@ -140,6 +140,7 @@ struct array_reduce {
 
   KOKKOS_INLINE_FUNCTION
   array_reduce &operator=(const array_reduce &src) {
+    if (&src == this) return *this;
```

Comment:
This is the change that caused failures on older ROCm versions.
@Rombur would you please look into this?

Reply:
ok

Reply:
I am not sure what's happening exactly but the compiler is definitely doing something weird. I can "fix" the test by adding a printf before the `if`. The issue has been fixed in ROCm 5.6. Since we are not using `clang-tidy` on the older ROCm version and the bug has been fixed, I propose that we guard the `if` for older  ROCm versions.

Reply:
Sure

Reply:
Will you take care of it?